agent/php_newrelic.h

```diff
@@ -503,6 +510,9 @@ nrtxn_t* txn; /* The all-important transaction pointer */
 char* predis_ctx; /* The current Predis pipeline context name, if any */
 nr_hashmap_t* predis_commands;
 
+nrcallbackfn_t* error_group_user_callback; /* The user defined callback for
+                                              error group naming */
+
```

Comment:
Why does this have to be a pointer to allocated memory and not a plain variable without dynamic memory management overhead?

Reply:
It's easier to check if it's been set or not simply by checking `NULL != error_group_user_callback`. Since it is more likely to not be set, I prefer the need to handle memory management over the need for a more involved check to see if the struct is set or not.

Reply:
Why not add a `is_set` flag to `nrcallbackfn_t` type? This would achieve the same without the memory management overhead.

Reply:
> When error group user callback is used, isn't it set for every request?
No. This will only ever be allocated if the API is called, and only deallocated if set (which can only happen if the API is called). Users who do not call the API will not be exposed to any of this functionality.

Reply:
So for an application that uses the error grouping callback, and sets it up early in the request processing, the memory will be allocated and freed for every request. Maybe it is a rare use case, but I'm just pointing out possible issues if dynamic memory is used for storing error grouping callback - performance of calling malloc/free and heap fragmentation in a long running process.

Reply:
I think the point about heap fragmentation is valid- I made the changes to implement this in 16ce1ffc2772b5a7d25c57ff7c1a246fc1e13be1.